File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/elementcache/VideoFormat.java
package com.gigigo.orchextra.core.domain.entities.elementcache;

import java.io.Serializable;

public enum VideoFormat implements Serializable {
  YOUTUBE("youtube"),
  VIMEO("vimeo"),
  NONE("");

  private final String videoFormat;

  VideoFormat(String videoFormat) {
    this.videoFormat = videoFormat;
  }

  public static VideoFormat convertStringToType(String format) {
    VideoFormat[] values = VideoFormat.values();
    for (VideoFormat value : values) {
      if (value.getVideoFormat().equals(format)) {
        return value;
      }
    }
    return NONE;
  }

  public String getVideoFormat() {
    return videoFormat;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/elementcache/VideoFormat.kt
package com.gigigo.orchextra.core.domain.entities.elementcache

import java.io.Serializable

enum class VideoFormat constructor(val videoFormat: String) : Serializable {
  YOUTUBE("youtube"),
  VIMEO("vimeo"),
  NONE("");

  companion object {

    fun convertStringToType(format: String): VideoFormat {
      for (value in VideoFormat.values()) {
        if (value.videoFormat.equals(format, true)) {
          return value
        }
      }
      return NONE
    }
  }
}